home/views.py

```diff
@@ -1,24 +1,35 @@
-from django.shortcuts import render
-from articles.models import Article, Tag
+from django.shortcuts import get_object_or_404, render
+from articles.models import Tag
 from django.contrib.auth import login as auth_login, authenticate
 from django.http import HttpResponseRedirect
 from django.contrib.auth.views import LoginView as AuthLoginView
 from django.shortcuts import redirect
-from django.db.models import Count
 from .forms import UserCreationForm, UserLoginForm
 
 
 def index(request):
-    articles = (
-        Article.objects.all()
-        .select_related("author__user")
-        .annotate(favorited_by__count=Count("favorited_by"))
-    )
     tags = Tag.objects.all()
+    tag_slug = request.GET.get("tag")
+    tag = None
+    active_tab = request.GET.get("active_tab", "global")
+    article_params = ""
+    if tag_slug:
+        tag = get_object_or_404(Tag, slug=tag_slug)
+        active_tab = "tag"
+        article_params = f"tag={tag_slug}"
+    if active_tab == "feed":
+        article_params = f"followed_by={request.user.username}"
```

Comment:
The tag selector is unique to the index page, but can control what is shown in the list frame. It does this by adding a query param to the index URL and reloading the page.

Reply:
The index page also has 2 tabs, one for `My Feed` and one for `Global`, also activated by query params.